script/gn-check.js

```diff
@@ -0,0 +1,23 @@
+const sh = require('shelljs')
+const path = require('path')
+const fs = require('fs')
+
+const ELECTRON_DIR = path.resolve(__dirname, '..', '..')
+let GN_CHECK_DIR
+
+if (process.env.ELECTRON_OUT_DIR) {
+  GN_CHECK_DIR = `../out/${process.env.ELECTRON_OUT_DIR}`
+} else {
+  for (const buildType of ['Debug', 'Testing', 'Release']) {
+    const outPath = path.resolve(ELECTRON_DIR, 'out', buildType)
+    if (fs.existsSync(outPath)) {
+      GN_CHECK_DIR = `../out/${buildType}`
+      break
+    }
+  }
+}
```

Comment:
Does it matter that our method of picking a fallback directory iff `ELECTRON_OUT_DIR` isn't set differs here from other parts of our `script/`directory?

It might make sense to port `get_out_dir()` from `script/lib/util.py` to `script/lib/utils.js` and use that ported function here (and also in `utils.getElectronExec()`)

Reply:
`get_out_dir()` is not used by us really though, it's all in CI, whereas most of us use testing builds 🤔

Reply:
The logic of `get_out_dir()` is also what's used by `utils.getElectronExec()`, which we do use for our testing builds' spec runs though?

I'm not against having one set of rules for human runs and another set for CI runs; I just don't want to introduce inconsistency for the point of inconsistency. If this new behavior is preferable, maybe that's what we should be doing in `utils.getElectronExec()` too

Reply:
hmm, ok!

Reply:
hmm, so `utils.getElectronExec()` just...creates `out/Debug` if none exists, which i don't feel is ideal behavior. I'd also rather change it, but perhaps in a follow-up PR?

Reply:
if there's perhaps a cleaner way to evaluate descending out dir type i'd be in favor of that though 😁

Reply:
Resolving the inconsistencies in a followup sounds good to me.

Reply:
https://github.com/electron/electron/pull/19857 🌟